### DISCUSSION TITLE
Need Help in trying out the Cluster Resource Set feature

### DISCUSSION THREAD

**Discussion by Sir-gits-a-lot** [kubernetes-sigs/cluster-api#8998] [Q&A] [answered]
Actually, I  wanted to try out the Cluster Resource Set experimental feature to automatically deploy CNI/CSI component to the workload cluster, but I am not able to apply the yaml manifest for which reference is provided in the capi book page. This is the reference link https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20200220-cluster-resource-set.md.
It's always throwing this error, error when creating "ClusterResourceSet.yaml": ClusterResourceSet in version "v1beta1" cannot be handled as a ClusterResourceSet: strict decoding error: unknown field "spec.mode".

If I change the version in the yaml manifest for Cluster Resource Set, it says that error: resource mapping not found for name: "crs1" namespace: "default" from "ClusterResourceSet.yaml": no matches for kind "ClusterResourceSet" in version "addons.cluster.x-k8s.io/v1"
ensure CRDs are installed first. 

I don't know how I can test out this Cluster Resource Set feature. If anyone has any clue, please let me know. Thanks.

**Comment by killianmuldoon** [comment, marked as answer]
It looks like the proposal - which the CAPI book links to - is out of date and not a good guide to creating a ClusterResourceSet. 

There's a better example in the CAPI repo which we use - after substitution - for the CAPI e2e tests.

**Comment by killianmuldoon** [comment]
Could you create an issue to improve the documentation around this?
